doc/develop/tools/clion.rst

```diff
@@ -0,0 +1,270 @@
+
+
+CLion
+#####
+
+`CLion <https://www.jetbrains.com/clion/>`_ is a cross-platform C/C++ IDE that
+supports multi-threaded RTOS debugging.
+
+This guide describes the process of setting up, building, and debugging
+Zephyr's :ref:`hello_world` sample in CLion.
+
+Get CLion
+*********
+
+Download_ and install CLion.
+
+.. _Download: https://www.jetbrains.com/clion/download
+
+Initialize a new workspace
+**************************
+
+This guide gives details on how to build and debug the
+:ref:`hello_world` sample application, but the instructions would
+be similar for any Zephyr project and
+:ref:`workspace layout west-workspaces`.
+
+Before you start, make sure you have a working Zephyr development
+environment, as per the instructions in the :ref:`getting_started`.
+
+Open the project in CLion
+**************************
+
+#. In CLion, click **Open** on the Welcome screen or select **File | Open**
+   from the main menu.
+
+#. Locate your Zephyr workspace folder (``~/zephyrproject`` if you
+    followed the Getting Started instructions) and select it.
+
+   Click **OK**.
+
+#. If prompted, click **Trust Project**.
+
+   See `this section`_ in CLion web help for more information on project
+   security.
+
+   .. _this section: https://www.jetbrains.com/help/clion/project-security.html#projects_security
+    </chapter>
+
+#. If the toolchain is not yet configured, CLion will open the
+    Open Project Wizard. We recommend that you use MinGW with default
+    settings on Windows or the System toolchain on Unix machines.
+
+
+Create a CMake presets file
+***************************
+
+This presets_ file will activate the Python virtual environment
+and will define the target board for CMake builds.
+
+.. _presets: https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html
+
+You can find more information on how CLion works with CMake presets on
+`this web help page`_.
+
+.. _this web help page: https://www.jetbrains.com/help/clion/cmake-presets.html
+
+#. In the **Project** view, locate the
+    ``zephyrproject\zephyr\samples\hello_world`` folder.
+
+#. Right-click the ``hello_world`` folder and select **New | File**.
+
+#. Name the file ``CMakeUserPresets.json``.
+
+#. Add the following code into it:
+
+   .. code-block::
+      :emphasize-lines: 13,14
+      {
+         "version": 3,
+            "cmakeMinimumRequired": {
+            "major": 3,
+            "minor": 24
+         },
+         "configurePresets": [
+            {
+               "name": "Zephyr-Debug",
+               "generator": "Ninja",
+               "binaryDir": "build",
+               "cacheVariables": {
+                  "BOARD": "nucleo_f302r8",
+                  "CMAKE_PROGRAM_PATH": "<openocd_installation>;<home>/zephyrproject/.venv/Scripts"
+               }
+            }
+         ],
+         "buildPresets": [
+            {
+               "name": "Zephyr-Debug",
+               "configurePreset": "Zephyr-Debug"
+            }
+         ]
+      }
+
```

Comment:
I have tried this Configure Preset, but for me it generates 2 presets: 1 Zephyr-Debug and a Zephyr-Debug-Zephyr-Debug
First one does not have "Build options" set second one has a "Build options": "--preset "Zephyr-Debug""

If I want some overlays and or config settings in these presets, how should I add these?

Reply:
Would it also be an option to add a preset for native_posix for example. This to see how you should work with multiple platforms. Would be a big plus for me. I can help with testing and trying out if you want.
I personally would love to get it working for nrf53 and posix. Always have been struggling with CLion.

Reply:
Do I understand correctly that `native_posix` and `native_posix_64` are not supported on windows?

Reply:
@MarinaKalashina I'd also add `.gitignore`:
```gitignore
/cmake-build*/
/build*/
CMakeUserPresets.json
```